### DISCUSSION TITLE
Checking input and applying it?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [bitbrain/beehave#278] [Q&A]
How do I check input and apply it via behavior trees?

I don't know how I would compound actions into things like axes and vectors

**Comment by bitbrain** [comment]
Any node has an `_input` method you could hook into. Since actions are nodes, you can do it on your custom action like so:
```
var event

func _input(event):
   this.event = event

func tick(actor, blackboard):
   if event != null:
      # handle latest input event here. Keep in mind that once it is handled, set it to `null`
```
The above is not optimal, as it can cause some problems with when some other node handles the event instead. Alternatively, you could also have a separate `InputHandler` in your scene that has the `Blackboard` as a dependency. This one then could set flags on the blackboard depending on certain inputs, and your actions then can access the blackboard.
